src/app/core/entity-components/entity-subrecord/entity-subrecord.component.spec.ts

```diff
@@ -33,4 +47,91 @@ describe("EntitySubrecordComponent", () => {
   it("should create", () => {
     expect(component).toBeTruthy();
   });
+
+  it("displays empty text instead of 'undefined' in textbox", () => {
+    component.records = [
+      Object.assign(new Entity(), { text: "foo", textarea: "foo" }),
+      Object.assign(new Entity(), { text: undefined, textarea: undefined }),
+    ];
+
+    component.columns = [
+      {
+        name: "text",
+        label: "Test text",
+        inputType: ColumnDescriptionInputType.TEXT,
+      },
+      {
+        name: "textarea",
+        label: "Test textarea",
+        inputType: ColumnDescriptionInputType.TEXTAREA,
+      },
+    ];
+    component.ngOnChanges({
+      records: new SimpleChange(undefined, component.records, true),
+      columns: new SimpleChange(undefined, component.columns, true),
+    });
+
+    component.recordsEditing.set(component.records[0].getId(), true);
+    component.recordsEditing.set(component.records[1].getId(), true);
+    fixture.detectChanges();
+
+    const inputFields = fixture.debugElement.queryAll(By.css("input"));
+    expect(inputFields[0].nativeElement.value).toBe("foo");
+    expect(inputFields[1].nativeElement.value).toBe("");
+
+    const textareaFields = fixture.debugElement.queryAll(By.css("textarea"));
+    expect(textareaFields[0].nativeElement.value).toBe("foo");
+    expect(textareaFields[1].nativeElement.value).toBe("");
+  });
+
+  it("apply formatter function to values in read mode if formatter is given", () => {
+    component.records = [
+      Object.assign(new Entity(), { simple: 0.9, percent: 0.9 }),
+    ];
+
+    component.columns = [
+      {
+        name: "simple",
+        label: "Test simple",
+        inputType: ColumnDescriptionInputType.NUMBER,
+      },
+      {
+        name: "percent",
+        label: "Test formatted",
+        inputType: ColumnDescriptionInputType.NUMBER,
+        valueFunction: (entity) => entity["percent"] * 100 + "%",
+      },
+    ];
+    component.ngOnChanges({
+      records: new SimpleChange(undefined, component.records, true),
+      columns: new SimpleChange(undefined, component.columns, true),
+    });
+    fixture.detectChanges();
+
+    const tdColumns = fixture.debugElement.queryAll(By.css("td"));
+    expect(tdColumns[0].nativeElement.innerText).toBe("0.9");
+    expect(tdColumns[1].nativeElement.innerText).toBe("90%");
+  });
+
+  it("formats MONTH automatically", () => {
```

Comment:
Maybe you could also add an test for the DAY, since it is also hard coded into the component.